src/Raven.Server/Smuggler/Documents/DatabaseDestination.cs

```diff
@@ -714,6 +719,40 @@ public async ValueTask DisposeAsync()
                             // waiting for the commands to be applied
                             await _database.RachisLogIndexNotifications.WaitForIndexNotification(_lastClusterTransactionIndex.Value, _token);
                         }
+                        else
+                        {
+                            Debug.Assert(_backupKind == BackupKind.Full || _backupKind == BackupKind.Incremental);
+                            ExecuteClusterTransactions();
+                        }
+                    }
+                }
+            }
+
+            private void ExecuteClusterTransactions()
+            {
+                long totalExecutedCommands;
+                using (_database.ServerStore.Engine.ContextPool.AllocateOperationContext(out ClusterOperationContext serverContext))
+                using (serverContext.OpenReadTransaction())
+                {
+                    var commandsCountPerDatabase = serverContext.Transaction.InnerTransaction.ReadTree(ClusterStateMachine.TransactionCommandsCountPerDatabase);
+                    var prevCount = ClusterTransactionCommand.GetPrevCount(serverContext, commandsCountPerDatabase, _database.Name);
+                    totalExecutedCommands = prevCount;
+                }
+
+                while (true)
+                {
+                    _token.ThrowIfCancellationRequested();
+                
+                    using (_database.ServerStore.ContextPool.AllocateOperationContext(out TransactionOperationContext serverContext))
+                    using (serverContext.OpenReadTransaction())
+                    {
+                        // the commands are already batched (10k or 16MB), so we are executing only 1 at a time
+                        var executed = _database.ExecuteClusterTransaction(serverContext, batchSize: 1);
+                        if (executed.Count == 0)
+                            break;
+                
+                        totalExecutedCommands += executed.Sum(x => x.Commands.Count);
+                        _result.AddInfo($"Executed {totalExecutedCommands:#,#;;0} cluster transaction commands.");
```

Comment:
`OnProgress` is missing here
https://github.com/ravendb/ravendb/pull/18339/files#diff-b8f4b8496625c671d24184474c3ee5e9aa1bf20c4859ab8fa5959da7073ffa57L837

Reply:
You use it here after that:
https://github.com/ravendb/ravendb/blob/3bd416cc5d472653dc0a45e0fa06eeb7f0fb6228/src/Raven.Server/Smuggler/Documents/DatabaseSmuggler.cs#L291
trying to pass here the 'OnProgress' would be a bit messy, what do you think?

Reply:
Done, 
also on the 6.0 PR:
https://github.com/ravendb/ravendb/pull/18323